subsys/net/ip/net_pkt.c

```diff
@@ -1245,6 +1245,10 @@ struct net_buf *net_frag_read(struct net_buf *frag, u16_t offset,
 		goto error;
 	}
 
+	if (len == 0) {
```

Comment:
I'm either in very bad state re: thinking, or: for len=0, `len--` in the while loop decrements len (which wraps around), but returns its original value, which is 0. Thus, the while condition is false, so the body doesn't execute and execution proceeds to `return frag;`. This is fully equivalent to the `if` you added. Was there a practical issue with how this func handles len=0?